### PR TITLE
refactor: change `WithValue` key type to any

### PR DESCRIPTION
**Pull request by mdanialr**
## 📑 Description

Implementation of [#630](https://github.com/goravel/framework/pull/630)

## Summary by CodeRabbit

- **New Features**
	- Enhanced context handling allows for greater flexibility with key types.
	- Improved usability of context methods for storing and retrieving values.
	- Updated HTTP route handlers to access context values without type assertions, increasing robustness.

- **Bug Fixes**
	- Updated dependency to a more recent version, potentially addressing existing issues.

- **Tests**
	- Expanded test coverage for context functionality, verifying the handling of various key types.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

**Comment by mdanialr** [comment]
@coderabbitai review

**Comment by mdanialr** [comment]
https://github.com/coderabbitai review

File: context.go [hunk 1]
```diff
@@ -61,11 +59,7 @@ func (c *Context) Err() error {
 }
 
 func (c *Context) Value(key any) any {
-	if keyStr, ok := key.(string); ok {
-		return c.instance.UserContext().Value(ctxKey(keyStr))
-	}
-
-	return nil
+	return c.instance.UserContext().Value(key)
```
[the remark this answers is not in the document]
Comment (mdanialr) [reply context.go:62]: done in https://github.com/goravel/fiber/pull/103/commits/b764dfe420397403e755e0693705f209a2073b0a

File: context_test.go [hunk 2]
```diff
@@ -7,10 +7,26 @@ import (
...
+	assert.Equal(t, "two point two", httpCtx.Value(2.2))
 
-	assert.Equal(t, httpCtx.Value("Hello").(string), "world")
-	assert.Equal(t, httpCtx.Value("Hi").(string), "Goravel")
+	ctx := httpCtx.Context()
+	assert.Equal(t, "world", ctx.Value("Hello"))
+	assert.Equal(t, "Goravel", ctx.Value("Hi"))
+	assert.Equal(t, "halo", ctx.Value(customKey))
+	assert.Equal(t, "one", ctx.Value(1))
+	assert.Equal(t, "two point two", ctx.Value(2.2))
```
Comment (hwbrzzl) [inline context_test.go:31]: 👍

**Review by hwbrzzl — APPROVED** [review]
Perfect

**Pull request merged**